changelog_unreleased/api/17344.md

```diff
@@ -0,0 +1,20 @@
+#### Ignore files with `--check-ignore-pragma` (#17344 by @wnayes)
+
+Individual files can now opt out of formatting via `@noformat` or `@noprettier` "pragma" comments at the top of the file.
```

Comment:
Why we should have both `@noformat` and `@noprettier` ? I feel one would be enough.
(I personally prefer `@noprettier`)

Reply:
One way I've been thinking about it is
- `@noprettier` means Prettier specifically should not be used.
- `@noformat` means no kind of formatter should be used (of which Prettier is one).

They each can express different sort of intents. Other formatters would have to align on the `@noformat` concept for that to be effective, though.

The other practical factor here is simply that `requirePragma` had supported `@prettier` and `@format`, so this reflects that.

Reply:
> The other practical factor here is simply that requirePragma had supported @prettier and @format, so this reflects that.

Ah it makes sense to align it with `requirePragma`.